src/test/scala/org/broadinstitute/dsde/agora/server/webservice/ApiServiceSpec.scala

```diff
@@ -41,63 +40,86 @@ with AgoraTestData with AgoraDbTest {
-      assert(entities === brief(Seq(testEntity1, testEntity2, testEntity3, testEntity4, testEntity5)))
-    }
+    Get(ApiUtil.Methods.withLeadingSlash + "?owner=" + agoraCIOwner.get + "&payload=" + uriEncode(payload1.get)) ~>
+      methodsService.queryRoute ~>
+      check {
+        handleError(
+          entity.as[Seq[AgoraEntity]],
+          (entities: Seq[AgoraEntity]) =>
+            assert(entities === brief(Seq(testEntity1, testEntity2, testEntity3, testEntity4, testEntity5)))
+        )
+      }
   }
 
   "Agora" should "create and return a method" in {
-    Post(ApiUtil.Methods.withLeadingSlash, testAgoraEntity) ~> methodsService.postRoute ~> check {
-      val agoraEntity = handleDeserializationErrors(entity.as[AgoraEntity])
-      assert(agoraEntity.namespace === namespace1)
-      assert(agoraEntity.name === name1)
-      assert(agoraEntity.synopsis === synopsis1)
-      assert(agoraEntity.documentation === documentation1)
-      assert(agoraEntity.owner === owner1)
-      assert(agoraEntity.payload === payload1)
-      assert(agoraEntity.snapshotId !== None)
-      assert(agoraEntity.createDate != null)
+    Post(ApiUtil.Methods.withLeadingSlash, testAgoraEntity) ~>
+      methodsService.postRoute ~> check {
+      handleError(entity.as[AgoraEntity], (entity: AgoraEntity) => {
+        assert(entity.namespace === namespace1)
+        assert(entity.name === name1)
+        assert(entity.synopsis === synopsis1)
+        assert(entity.documentation === documentation1)
+        assert(entity.owner === agoraCIOwner)
+        assert(entity.payload === payload1)
+        assert(entity.snapshotId !== None)
+        assert(entity.createDate != null)
+      }
+      )
     }
   }
 
   "Agora" should "return a 400 bad request when posting a malformed payload" in {
-    Post(ApiUtil.Methods.withLeadingSlash, testBadAgoraEntity) ~> methodsService.postRoute ~> check {
+    Post(ApiUtil.Methods.withLeadingSlash, testBadAgoraEntity) ~>
+      methodsService.postRoute ~> check {
       assert(status === BadRequest)
       assert(responseAs[String] != null)
     }
   }
 
   "Agora" should "store 10kb of github markdown as method documentation and return it without alteration" in {
-    Post(ApiUtil.Methods.withLeadingSlash, testAgoraEntityBigDoc) ~> methodsService.postRoute ~> check {
-      val agoraEntity = handleDeserializationErrors(entity.as[AgoraEntity])
-      assert(agoraEntity.documentation === bigDocumentation)
+    Post(ApiUtil.Methods.withLeadingSlash, testAgoraEntityBigDoc) ~>
+      methodsService.postRoute ~> check {
+      handleError(
+        entity.as[AgoraEntity],
+        (entity: AgoraEntity) => assert(entity.documentation.get === bigDocumentation)
+      )
     }
   }
 
-  def handleDeserializationErrors[T](deserialized: Deserialized[T]): T = {
-    if (deserialized.isRight) deserialized.right.get else failTest(deserialized.left.get.toString)
+  def handleError[T](deserialized: Deserialized[T], assertions: (T) => Unit) = {
```

Comment:
Assertion failures seem to be ignored when wrapped in handleError(). For example if I try adding assert(false === true) into the tests above, they still pass.